src/main/java/edu/kis/powp/jobs2d/drivers/DriverManager.java

```diff
@@ -10,12 +10,14 @@
 public class DriverManager {
 
     private Job2dDriver currentDriver = new LoggerDriver();
+    private DriverStatistics currentStatistics = new DriverStatistics();
 
     /**
      * @param driver Set the driver as current.
      */
-    public synchronized void setCurrentDriver(Job2dDriver driver) {
+    public synchronized void setCurrentDriver(Job2dDriver driver, DriverStatistics statistics) {
```

Comment:
Please do not force operating on drivers together with statistics (mixing responsibilities here). When "Observer for a driver changes events" task will be finished you will be able to achieve similar functionality without tangling the implementation.